crates/interpreter/src/valid.rs

```diff
@@ -514,7 +536,8 @@ enum LabelKind<'m> {
 impl<'a, 'm> Expr<'a, 'm> {
     fn new(
         context: &'a Context<'m>, parser: &'a mut Parser<'m>,
-        is_const: Result<&'a mut [bool], &'a [bool]>,
+        is_const: Result<&'a mut [bool], &'a [bool]>, type_idx: Option<TypeIdx>,
+        parser_range: ParserRange,
```

Comment:
It seems to me we don't need `type_idx` and `parser_range` for `Expr` to function properly. So instead of rename `SideTable` to `FuncMeta`, we just rename it to `BranchTable` with just the one field it used to have. Then in `check_module()`, when going through the code section, we get the output of `check_body()` to create a `FuncMeta` using the `type_idx` and `parser_range` information we have there.